lib/CMakeLists.txt

```diff
@@ -4,6 +4,11 @@ if(USE_BUNDLED_TINYXML2)
     include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/externals/tinyxml2/)
 endif()
 include_directories(${PROJECT_SOURCE_DIR}/externals/simplecpp/)
+if (USE_BOOST_MULTIPREC)
+    find_package(Boost REQUIRED COMPONENTS system)
+    include_directories(${Boost_INCLUDE_DIRS})
+    add_compile_definitions(USE_BOOST_MULTIPREC)
```

Comment:
This should use: 

```cmake
target_include_directories(lib_objs PUBLIC SYSTEM ${Boost_INCLUDE_DIRS})
target_compile_definitions(lib_objs PUBLIC USE_BOOST_MULTIPREC)
```

Reply:
I guess this makes it specific to lib_objs?

Reply:
That's `add_definitions(-DUSE_BOOST_MULTIPREC)` - `target_compile_definitions()` is not available in CMake 2.8 (let's tackle a CMake update later - if I ever get healthy again).

Also those should be placed in `findDependencies.cmake` and `compilerDefinitions.cmake`.

Also those should be `PRIVATE` so they don't spill into everything.